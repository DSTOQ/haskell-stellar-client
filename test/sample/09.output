TransactionEnvelope
  { _transaction =
      Transaction
        { _sourceAccount =
            PublicKeyEd25519
              { _publicKeyEd25519 =
                  [ 102
                  , 97
                  , 48
                  , 55
                  , 56
                  , 50
                  , 57
                  , 97
                  , 53
                  , 49
                  , 52
                  , 50
                  , 98
                  , 54
                  , 54
                  , 97
                  , 51
                  , 57
                  , 57
                  , 97
                  , 53
                  , 102
                  , 101
                  , 97
                  , 51
                  , 55
                  , 97
                  , 102
                  , 98
                  , 102
                  , 98
                  , 53
                  , 53
                  , 50
                  , 97
                  , 101
                  , 56
                  , 51
                  , 57
                  , 55
                  , 55
                  , 50
                  , 51
                  , 48
                  , 98
                  , 50
                  , 97
                  , 100
                  , 99
                  , 52
                  , 49
                  , 50
                  , 99
                  , 55
                  , 98
                  , 55
                  , 99
                  , 52
                  , 102
                  , 101
                  , 54
                  , 51
                  , 53
                  , 49
                  ]
              }
        , _fee = Fee { _fee = 500 }
        , _seqNum = SequenceNumber { _sequenceNumber = 46283194646921285 }
        , _timeBounds = Nothing
        , _memo = MemoNone
        , _operations =
            [ Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling = AssetNative
                        , _buying =
                            AssetCreditAlphanum
                              AssetCode { _assetCode = "CNY" }
                              PublicKeyEd25519
                                { _publicKeyEd25519 =
                                    [ 49
                                    , 97
                                    , 57
                                    , 51
                                    , 50
                                    , 52
                                    , 50
                                    , 54
                                    , 53
                                    , 56
                                    , 97
                                    , 99
                                    , 102
                                    , 56
                                    , 49
                                    , 56
                                    , 101
                                    , 49
                                    , 49
                                    , 99
                                    , 98
                                    , 97
                                    , 48
                                    , 56
                                    , 51
                                    , 55
                                    , 53
                                    , 56
                                    , 52
                                    , 102
                                    , 52
                                    , 55
                                    , 53
                                    , 55
                                    , 57
                                    , 50
                                    , 50
                                    , 53
                                    , 52
                                    , 49
                                    , 97
                                    , 55
                                    , 54
                                    , 55
                                    , 55
                                    , 101
                                    , 57
                                    , 57
                                    , 50
                                    , 98
                                    , 50
                                    , 49
                                    , 50
                                    , 49
                                    , 55
                                    , 54
                                    , 97
                                    , 49
                                    , 57
                                    , 101
                                    , 99
                                    , 48
                                    , 101
                                    , 51
                                    ]
                                }
                        , _amount = 10000000
                        , _price = Price { _numerator = 703 , _denominator = 1000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling = AssetNative
                        , _buying =
                            AssetCreditAlphanum
                              AssetCode { _assetCode = "USD" }
                              PublicKeyEd25519
                                { _publicKeyEd25519 =
                                    [ 49
                                    , 97
                                    , 57
                                    , 51
                                    , 50
                                    , 52
                                    , 50
                                    , 54
                                    , 53
                                    , 56
                                    , 97
                                    , 99
                                    , 102
                                    , 56
                                    , 49
                                    , 56
                                    , 101
                                    , 49
                                    , 49
                                    , 99
                                    , 98
                                    , 97
                                    , 48
                                    , 56
                                    , 51
                                    , 55
                                    , 53
                                    , 56
                                    , 52
                                    , 102
                                    , 52
                                    , 55
                                    , 53
                                    , 55
                                    , 57
                                    , 50
                                    , 50
                                    , 53
                                    , 52
                                    , 49
                                    , 97
                                    , 55
                                    , 54
                                    , 55
                                    , 55
                                    , 101
                                    , 57
                                    , 57
                                    , 50
                                    , 98
                                    , 50
                                    , 49
                                    , 50
                                    , 49
                                    , 55
                                    , 54
                                    , 97
                                    , 49
                                    , 57
                                    , 101
                                    , 99
                                    , 48
                                    , 101
                                    , 51
                                    ]
                                }
                        , _amount = 10000000
                        , _price = Price { _numerator = 293 , _denominator = 625 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling = AssetNative
                        , _buying =
                            AssetCreditAlphanum
                              AssetCode { _assetCode = "XRP" }
                              PublicKeyEd25519
                                { _publicKeyEd25519 =
                                    [ 49
                                    , 97
                                    , 57
                                    , 51
                                    , 50
                                    , 52
                                    , 50
                                    , 54
                                    , 53
                                    , 56
                                    , 97
                                    , 99
                                    , 102
                                    , 56
                                    , 49
                                    , 56
                                    , 101
                                    , 49
                                    , 49
                                    , 99
                                    , 98
                                    , 97
                                    , 48
                                    , 56
                                    , 51
                                    , 55
                                    , 53
                                    , 56
                                    , 52
                                    , 102
                                    , 52
                                    , 55
                                    , 53
                                    , 55
                                    , 57
                                    , 50
                                    , 50
                                    , 53
                                    , 52
                                    , 49
                                    , 97
                                    , 55
                                    , 54
                                    , 55
                                    , 55
                                    , 101
                                    , 57
                                    , 57
                                    , 50
                                    , 98
                                    , 50
                                    , 49
                                    , 50
                                    , 49
                                    , 55
                                    , 54
                                    , 97
                                    , 49
                                    , 57
                                    , 101
                                    , 99
                                    , 48
                                    , 101
                                    , 51
                                    ]
                                }
                        , _amount = 70000000
                        , _price = Price { _numerator = 2171 , _denominator = 10000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling = AssetNative
                        , _buying =
                            AssetCreditAlphanum
                              AssetCode { _assetCode = "XYG" }
                              PublicKeyEd25519
                                { _publicKeyEd25519 =
                                    [ 49
                                    , 97
                                    , 57
                                    , 51
                                    , 50
                                    , 52
                                    , 50
                                    , 54
                                    , 53
                                    , 56
                                    , 97
                                    , 99
                                    , 102
                                    , 56
                                    , 49
                                    , 56
                                    , 101
                                    , 49
                                    , 49
                                    , 99
                                    , 98
                                    , 97
                                    , 48
                                    , 56
                                    , 51
                                    , 55
                                    , 53
                                    , 56
                                    , 52
                                    , 102
                                    , 52
                                    , 55
                                    , 53
                                    , 55
                                    , 57
                                    , 50
                                    , 50
                                    , 53
                                    , 52
                                    , 49
                                    , 97
                                    , 55
                                    , 54
                                    , 55
                                    , 55
                                    , 101
                                    , 57
                                    , 57
                                    , 50
                                    , 98
                                    , 50
                                    , 49
                                    , 50
                                    , 49
                                    , 55
                                    , 54
                                    , 97
                                    , 49
                                    , 57
                                    , 101
                                    , 99
                                    , 48
                                    , 101
                                    , 51
                                    ]
                                }
                        , _amount = 40000000
                        , _price = Price { _numerator = 1831 , _denominator = 2000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling =
                            AssetCreditAlphanum
                              AssetCode { _assetCode = "XLM" }
                              PublicKeyEd25519
                                { _publicKeyEd25519 =
                                    [ 49
                                    , 97
                                    , 57
                                    , 51
                                    , 50
                                    , 52
                                    , 50
                                    , 54
                                    , 53
                                    , 56
                                    , 97
                                    , 99
                                    , 102
                                    , 56
                                    , 49
                                    , 56
                                    , 101
                                    , 49
                                    , 49
                                    , 99
                                    , 98
                                    , 97
                                    , 48
                                    , 56
                                    , 51
                                    , 55
                                    , 53
                                    , 56
                                    , 52
                                    , 102
                                    , 52
                                    , 55
                                    , 53
                                    , 55
                                    , 57
                                    , 50
                                    , 50
                                    , 53
                                    , 52
                                    , 49
                                    , 97
                                    , 55
                                    , 54
                                    , 55
                                    , 55
                                    , 101
                                    , 57
                                    , 57
                                    , 50
                                    , 98
                                    , 50
                                    , 49
                                    , 50
                                    , 49
                                    , 55
                                    , 54
                                    , 97
                                    , 49
                                    , 57
                                    , 101
                                    , 99
                                    , 48
                                    , 101
                                    , 51
                                    ]
                                }
                        , _buying = AssetNative
                        , _amount = 10000000
                        , _price = Price { _numerator = 1003 , _denominator = 2000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            ]
        }
  , _signatures =
      [ DecoratedSignature
          { _hint = SignatureHint [ 99 , 52 , 102 , 101 , 54 , 51 , 53 , 49 ]
          , _signature =
              Signature
                { _signature =
                    "[48,52,102,98,51,50,53,52,51,50,98,52,51,57,101,52,52,97,52,57,51,49,101,102,100,53,100,55,49,99,102,102,49,98,51,100,49,97,53,102,52,55,99,52,51,51,101,51,97,48,52,48,54,98,56,52,98,54,54,51,101,56,54,57,52,51,54,101,55,48,57,55,51,98,55,102,51,54,98,100,99,53,56,101,49,57,55,102,100,50,97,52,49,97,54,101,49,102,51,55,55,100,48,52,102,101,48,56,54,56,50,54,48,99,52,53,100,53,49,51,99,102,99,98,50,97,48,51]"
                }
          }
      ]
  }