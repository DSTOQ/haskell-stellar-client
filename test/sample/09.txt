AAAAAPoHgppRQrZqOZpf6jevv7VSroOXcjCyrcQSx7fE/mNRAAAB9ACkblIAAABFAAAAAAAAAAAAAAAFAAAAAAAAAAMAAAAAAAAAAUNOWQAAAAAAGpMkJlis+BjhHLoIN1hPR1eSJUGnZ36ZKyEhdqGewOMAAAAAAJiWgAAAAr8AAAPoAAAAAAAAAAAAAAAAAAAAAwAAAAAAAAABVVNEAAAAAAAakyQmWKz4GOEcugg3WE9HV5IlQadnfpkrISF2oZ7A4wAAAAAAmJaAAAABJQAAAnEAAAAAAAAAAAAAAAAAAAADAAAAAAAAAAFYUlAAAAAAABqTJCZYrPgY4Ry6CDdYT0dXkiVBp2d+mSshIXahnsDjAAAAAAQsHYAAAAh7AAAnEAAAAAAAAAAAAAAAAAAAAAMAAAAAAAAAAVhZRwAAAAAAGpMkJlis+BjhHLoIN1hPR1eSJUGnZ36ZKyEhdqGewOMAAAAAAmJaAAAABycAAAfQAAAAAAAAAAAAAAAAAAAAAwAAAAFYTE0AAAAAABqTJCZYrPgY4Ry6CDdYT0dXkiVBp2d+mSshIXahnsDjAAAAAAAAAAAAmJaAAAAD6wAAB9AAAAAAAAAAAAAAAAAAAAABxP5jUQAAAEAE+zJUMrQ55EpJMe/V1xz/Gz0aX0fEM+OgQGuEtmPoaUNucJc7fza9xY4Zf9KkGm4fN30E/ghoJgxF1RPPyyoD
