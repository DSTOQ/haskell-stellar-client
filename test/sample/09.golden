TransactionEnvelope
  { _transaction =
      Transaction
        { _sourceAccount =
            GD5APAU2KFBLM2RZTJP6UN5PX62VFLUDS5ZDBMVNYQJMPN6E7ZRVDYNI
        , _fee = FeeStroops { _feeStroops = 500 }
        , _seqNum = 46283194646921285
        , _timeBounds = Nothing
        , _memo = MemoNone
        , _operations =
            [ Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling = AssetNative
                        , _buying =
                            AssetCreditAlphanum
                              CNY GANJGJBGLCWPQGHBDS5AQN2YJ5DVPERFIGTWO7UZFMQSC5VBT3AOGH6P
                        , _amount = NonNegativeInt64 (Refined 10000000)
                        , _price = Price { _numerator = 703 , _denominator = 1000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling = AssetNative
                        , _buying =
                            AssetCreditAlphanum
                              USD GANJGJBGLCWPQGHBDS5AQN2YJ5DVPERFIGTWO7UZFMQSC5VBT3AOGH6P
                        , _amount = NonNegativeInt64 (Refined 10000000)
                        , _price = Price { _numerator = 293 , _denominator = 625 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling = AssetNative
                        , _buying =
                            AssetCreditAlphanum
                              XRP GANJGJBGLCWPQGHBDS5AQN2YJ5DVPERFIGTWO7UZFMQSC5VBT3AOGH6P
                        , _amount = NonNegativeInt64 (Refined 70000000)
                        , _price = Price { _numerator = 2171 , _denominator = 10000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling = AssetNative
                        , _buying =
                            AssetCreditAlphanum
                              XYG GANJGJBGLCWPQGHBDS5AQN2YJ5DVPERFIGTWO7UZFMQSC5VBT3AOGH6P
                        , _amount = NonNegativeInt64 (Refined 40000000)
                        , _price = Price { _numerator = 1831 , _denominator = 2000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            , Operation
                { _sourceAccount = Nothing
                , _body =
                    ManageOffer
                      ManageOfferOp
                        { _selling =
                            AssetCreditAlphanum
                              XLM GANJGJBGLCWPQGHBDS5AQN2YJ5DVPERFIGTWO7UZFMQSC5VBT3AOGH6P
                        , _buying = AssetNative
                        , _amount = NonNegativeInt64 (Refined 10000000)
                        , _price = Price { _numerator = 1003 , _denominator = 2000 }
                        , _offerId = OfferId { _offerId = 0 }
                        }
                }
            ]
        }
  , _signatures =
      [ DecoratedSignature
          { _hint = SignatureHint { _signatureHint = "c4fe6351" }
          , _signature =
              Signature
                { _signature =
                    "04fb325432b439e44a4931efd5d71cff1b3d1a5f47c433e3a0406b84b663e869436e70973b7f36bdc58e197fd2a41a6e1f377d04fe0868260c45d513cfcb2a03"
                }
          }
      ]
  }